### ISSUE TITLE
Early Stopping Training: ClassificationScoreCalculator not working

### ISSUE THREAD

**Issue by Aulig** [deeplearning4j/deeplearning4j#5459]
Im using a MultiLayerNetwork for classification. The early stopping trainer always stops the training after the number of epochs specified for epochs without improvement (new ScoreImprovementEpochTerminationCondition(maxNoImprovementEpochs)). The problem however is that the model considered the best by the early stopping trainer (epoch 0) is by no means better than the latest model. If i switch out the ClassificationScoreCalculator for a DataSetLossCalculator it acts as expected. Here is a single class model you can use to see it:

https://pastebin.com/Edvbg9Rz
(dataset is attached to this issue)
This is the stats it achieved for me using Metric.ACCURACY :

> Termination reason: EpochTerminationCondition
> Termination details: ScoreImprovementEpochTerminationCondition(maxEpochsWithNoImprovement=100, minImprovement=0.0)
> Total epochs: 101
> Best epoch number: 0
> Score at best epoch: 0.2467700258397933

> -----"Best" (According to earlystopping training) Model-----

> ---Testing Dataset---
>  Accuracy:        0,2468
 
> ---Training Dataset---
>  Accuracy:        0,2735

> -----Latest Model-----

> ---Testing Dataset---
>  Accuracy:        0,4910

> ---Training Dataset---
>  Accuracy:        0,4688

[minimalTraining.txt](https://github.com/deeplearning4j/deeplearning4j/files/2069909/minimalTraining.txt)
[minimalTesting.txt](https://github.com/deeplearning4j/deeplearning4j/files/2069910/minimalTesting.txt)

**Comment by AlexDBlack** [comment]
Thanks for reporting: this bug has been fixed here: https://github.com/deeplearning4j/deeplearning4j/pull/5565

You can access the fix by using snapshots: https://deeplearning4j.org/snapshots
